src/tools/fuzzing/heap-types.cpp

```diff
@@ -712,21 +713,13 @@ struct Inhabitator {
 Inhabitator::Variance Inhabitator::getVariance(FieldPos field) {
   auto [type, idx] = field;
   assert(!type.isBasic() && !type.isSignature());
-  if (type.isStruct()) {
-    if (type.getStruct().fields[idx].mutable_ == Mutable) {
-      return Invariant;
-    } else {
-      return Covariant;
-    }
-  }
-  if (type.isArray()) {
-    if (type.getArray().element.mutable_ == Mutable) {
-      return Invariant;
-    } else {
-      return Covariant;
-    }
+  auto field = GCTypeUtils::getField(type, idx);
+  assert(field);
+  if (field->.mutable_ == Mutable) {
```

Comment:
Whoa, I've never seen `->.` before. I'm surprised that compiles. Does it work if you just do `field->mutable_`?

Reply:
It didn't compile... somehow I forgot to check that this compiles before pushing it :smile:

Reply:
(should be ok now)